Editor/Scripts/Utilities.cs

```diff
@@ -12,7 +11,17 @@ public static class Utilities
         public static IEnumerable<Sprite> GetSpritesFromTexture(Object texture)
         {
             var path = AssetDatabase.GetAssetPath(texture);
-            var subAssets = AssetDatabase.LoadAllAssetsAtPath(path);
+            Object[] subAssets = AssetDatabase.LoadAllAssetsAtPath(path);
+
+            //Sort the Sprite by their name's trailing number.
+            subAssets = subAssets.OrderBy(x =>
+            {
+                string trailingNumber = string.Concat(x.name.ToArray().Reverse().TakeWhile(char.IsNumber).Reverse());
+                int index = 0;
+                int.TryParse(trailingNumber, out index);
+                return index;
+            }).ToArray();
```

Comment:
I mean why we cant just go with this?

```suggestion
            subAssets = subAssets.OrderBy(x => x.name).ToArray();
```

I believe this can be optimized as (need to test is this compiles)

```suggestion
            Array.Sort(subAssets, (x, y) => string.Compare(x.name, y.name));
```

Reply:
Ok,

Here's a screenshot of how Unity's default naming for the TileMapEditor sliced sprites. 
![2024-11-06_17-20](https://github.com/user-attachments/assets/7395a6f0-b6f5-4ba7-9205-cffd03d2da61)

The Sprites' naming at bottom right represent the sprites sliced in the SpriteEditor order from Top-Left to Bottom-Right. And this is how Unity name and order the sliced sprites.

Now please notice the trailing digits of the sprites' name. You cannot order them correctly simply by the default string compare sorting or Order() stuffs. 
![2024-11-06_17-40](https://github.com/user-attachments/assets/da434f71-0939-4946-91bd-e7b0d03216ed)

Here's the result my code can do when import a texture to the QuickRuleTileEditorWindow:
![2024-11-06_17-33](https://github.com/user-attachments/assets/657d7559-134a-4e3e-9238-d01549ca34ab)
Which show's the order is the same as what you see in the SpriteEditor window.

Here's the import result of the code you provided:
![2024-11-06_17-35](https://github.com/user-attachments/assets/72ea9fb1-d183-4b7d-95d1-9a99244d8cf8)

To achieve the desired result we'll need to try take the trailing number and order by respect them as a 'number'. And that's why my code did that.

Reply:
Oh, thats strange, I checked this in different environment (not unity) and it worked as expected, there is probably a way to order strings with other comparison method that will do a trick.

Reply:
Also, do you use `OrderBy` or `Array.Sort` in this example?

Reply:
Looks like there are ready to use function

```suggestion
           System.Array.Sort(subAssets, (x, y) => EditorUtility.NaturalCompare(x.name, y.name));
```

Reply:
Nice digging! That did the trick!

Reply:
BTW, it was OrderBy